hls4ml/model/layers.py

```diff
@@ -359,17 +362,44 @@ def initialize(self):
 
 
 class Reshape(Layer):
+    _expected_attributes = [
+        Attribute('target_shape', value_type=typing.Sequence),
+    ]
+
     def initialize(self):
-        shape = self.attributes['target_shape']
-        if shape[0] is None:
-            shape = shape[1:]
-        dims = [f'N_SIZE_{i}_{self.index}' for i in range(1, len(shape) + 1)]
+        input_shape = self.get_input_variable(self.inputs[0]).shape
+        target_shape = self.get_attr('target_shape')
+        if target_shape is None:
+            # need to get it from the input
+            shape_node = self.get_input_node(self.inputs[1])
+            # for QONNX, remove batch dimension
+            if shape_node:
+                target_shape = shape_node.value[1:]
+            else:
+                raise RuntimeError("Reshape for ONNX requires the target shape to be a second input.")
 
-        out_name = self.outputs[0]
-        proxy = self.get_input_variable()
-        out = InplaceVariable(shape, dims, proxy)
+        # remove Nones -- is this ever triggered?
```

Comment:
The way I see this failing is if you specify the batch size during the creation of the model (like passing `batch_size` to the `Input` layer that some RNNs do). Then the first element won't be None in the json of the model that we parse. Not sure if you would even get this far, or will the conversion fail somewhere earlier.

Reply:
In ONNX the batch dimension is not None, and from what I gather can be any number, not necessarily just 1. We need to have a robust way to tell what is the batch dimension that needs to be removed. (Alternately, it can be kept and ignored in all cases.)

Reply:
The number of elements must match, if it doesn't and the first dimension is not none, that's a batch. Check that is a multiple just to be sure. The number of elements can match, the first one can be 1 and be a batch, but I think in that case we should still be fine.